### PR TITLE
Add required tests for `go/mysql/hex`

### PR DESCRIPTION
**Pull request by beingnoble03**
## Description
This PR adds required tests for `go/mysql/hex`

## Related Issue(s)
Fixes part of #14931 

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/mysql/hex/hex_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		input []byte
+		want  []byte
+	}{
+		{[]byte{0xAB, 0xCD, 0xEF}, []byte("ABCDEF")},
+		{[]byte{0x01, 0x23, 0x45}, []byte("012345")},
+	}
+
+	for _, tCase := range testCases {
+		got := EncodeBytes(tCase.input)
+		assert.Equalf(t, tCase.want, got, "got %v, want %v", got, tCase.want)
```
Comment (dbussink) [inline go/mysql/hex/hex_test.go:36]: I think we can just do `assert.Equal(t, tCase.want, got)` here and in the other cases. The additional stuff here doesn't seem to add much.

This also applies to the other cases here in this file.

Comment (beingnoble03) [reply go/mysql/hex/hex_test.go:36]: I understand. Makes sense; I'll proceed accordingly.

File: go/mysql/hex/hex_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	err := DecodeBytes([]byte("testDst"), []byte("1"))
+	assert.NoError(t, err)
+
+	err = DecodeBytes([]byte("testDst"), []byte("12"))
+	assert.NoError(t, err)
+
+	// DecodeBytes should return an error for "é" as
+	// hex.decode returns an error for non-ASCII characters
+	err = DecodeBytes([]byte("testDst"), []byte("é"))
+	assert.Errorf(t, err, "expected error to appear for invalid byte, got %v", err)
```
Comment (dbussink) [inline go/mysql/hex/hex_test.go:101]: I think you can also drop the `Errorf` here. It's not very useful, as `err` will always be `nil` anyway.

Comment (beingnoble03) [reply go/mysql/hex/hex_test.go]: Hmm...right. Done. Replaced the error message with `DecodeBytes() should have errored as src contains non-ASCII character`.

File: go/mysql/hex/hex_test.go [hunk 3]
```diff
@@ -98,5 +98,5 @@ func TestDecodeBytes(t *testing.T) {
 	// DecodeBytes should return an error for "é" as
 	// hex.decode returns an error for non-ASCII characters
 	err = DecodeBytes([]byte("testDst"), []byte("é"))
-	assert.Error(t, err)
+	assert.Error(t, err, "DecodeBytes() should have errored as src contains non-ASCII character")
```
Comment (dbussink) [inline go/mysql/hex/hex_test.go:101]: Any reason for having a custom message at all?

Comment (beingnoble03) [reply go/mysql/hex/hex_test.go:101]: We have added custom messages earlier in other tests too. So, I added here.

Comment (dbussink) [reply go/mysql/hex/hex_test.go:101]: I don’t think it add a lot of value here? I think we can go for the minimal setup and only add something custom if there’s explicit value in doing so.

Comment (beingnoble03) [reply go/mysql/hex/hex_test.go]: Done.

**Review by dbussink — APPROVED** [review]

**Review by frouioui — APPROVED** [review]
Looks good to me!

**Pull request merged**